### PR TITLE
Translate "Ruby 3.4.0 released" (ko)

### PR DESCRIPTION
**Pull request by riseshia**
:link: https://github.com/ruby/www.ruby-lang.org/issues/2818

Translate of #3442 

Actual diff is 792b349

### CODE REVIEW

File: ko/news/_posts/2024-12-25-ruby-3-4-0-released.md [hunk 1]
```diff
@@ -0,0 +1,344 @@
...
+
+* 블록을 인덱스로 사용할 수 없게 됩니다. [[Bug #19918]]
+
+* 키워드 인수를 인덱스로 사용할 수 없게 됩니다. [[Bug #20218]]
+
+* 최상위 이름 `::Ruby`은 예약되었으며, 이를 정의할 경우 `Warning[:deprecated]`가 발생합니다. [[Feature #20884]]
+
+## 코어 클래스 변경
+
+주의: 눈에 띄는 클래스 변경만을 포함합니다.
```
Comment (yous) [inline ko/news/_posts/2024-12-25-ruby-3-4-0-released.md:125]:
```suggestion
주의: 눈에 띄는 코어 클래스 변경만을 포함합니다.
```

File: ko/news/_posts/2024-12-25-ruby-3-4-0-released.md [hunk 2]
```diff
@@ -0,0 +1,344 @@
...
+
+* Exception
+
+  * `Exception#set_backtrace`는 이제 `Thread::Backtrace::Location`의 배열을 입력으로 받을 수 있습니다.
+    `Kernel#raise`, `Thread#raise`와 `Fiber#raise`도 같은 형식의 입력을 받습니다. [[Feature #13557]]
+
+* GC
+
+    * `GC.config`가 추가되어 가비지 컬렉터(GC)에 설정을 변경할 수 있습니다.
```
Comment (yous) [inline ko/news/_posts/2024-12-25-ruby-3-4-0-released.md:134]:
```suggestion
    * `GC.config`가 추가되어 가비지 컬렉터(GC)의 설정을 변경할 수 있습니다.
```

File: ko/news/_posts/2024-12-25-ruby-3-4-0-released.md [hunk 3]
```diff
@@ -0,0 +1,344 @@
...
+      [[Feature #20443]]
+
+    * GC 설정 `rgengc_allow_full_mark`가 추가되었습니다. `false`일 때
```
Comment (yous) [inline ko/news/_posts/2024-12-25-ruby-3-4-0-released.md:137]:
```suggestion
    * GC 설정 파라미터 `rgengc_allow_full_mark`가 추가되었습니다. `false`일 때
```

File: ko/news/_posts/2024-12-25-ruby-3-4-0-released.md [hunk 4]
```diff
@@ -0,0 +1,344 @@
...
+      GC는 젊은 객체만 마킹합니다. 기본값은 `true`입니다. [[Feature #20443]]
+
+* Ractor
+
+    * Ractor 내부에서 `require`가 허용됩니다. 해당 처리는 주 Ractor에서
+      실행됩니다. 불러오는 처리를 주 Ractor 에서 실행하는
```
Comment (yous) [inline ko/news/_posts/2024-12-25-ruby-3-4-0-released.md:143]:
```suggestion
      실행됩니다. 불러오는 처리를 주 Ractor에서 실행하는
```

File: ko/news/_posts/2024-12-25-ruby-3-4-0-released.md [hunk 5]
```diff
@@ -0,0 +1,344 @@
...
+
+## 표준 라이브러리 변경
+
+주의: 눈에 띄는 표준 라이브러리 변경만을 포함합니다.
+
+* RubyGems
+    * `--attestation` 옵션을 gem push에 추가했습니다. [sigstore.dev]에 서명을 저장할 수 있습니다.
+
+* Bundler
+    * 새 lockfile 생성시에 체크섬을 포함하는 `lockfile_checksums` 설정을 추가합니다.
```
Comment (yous) [inline ko/news/_posts/2024-12-25-ruby-3-4-0-released.md:168]:
```suggestion
    * 새 lockfile 생성 시에 체크섬을 포함하는 `lockfile_checksums` 설정을 추가합니다.
```

File: ko/news/_posts/2024-12-25-ruby-3-4-0-released.md [hunk 6]
```diff
@@ -0,0 +1,344 @@
...
+
+`it`은 변수 이름 없이 블록 파라미터를 참조하기 위해 추가되었습니다. [[Feature #18980]]
+
+```ruby
+ary = ["foo", "bar", "baz"]
+
+p ary.map { it.upcase } #=> ["FOO", "BAR", "BAZ"]
+```
+
+`it`은 `_1`과 거의 동일하게 동작합니다. 블록에서 `_1`만 사용할 때 다른 번호 파라미터인 `_2` 등이 나타날 가능성이 읽는 사람에게  추가적인 인지 부담을 주게 됩니다. 따라서 `it`은 편리한 별칭으로 추가되었습니다. `it`은 `it` 자체로 충분히 명확한 경우에 사용하세요. 예를 들어, 인라인 블록에서 사용하세요.
```
[the remark this answers is not in the document]
Comment (riseshia) [reply ko/news/_posts/2024-12-25-ruby-3-4-0-released.md]: Lgtm!

File: ko/news/_posts/2024-12-25-ruby-3-4-0-released.md [hunk 7]
```diff
@@ -0,0 +1,344 @@
...
+
+## 소켓 라이브러리가 Happy Eyeballs Version 2 (RFC 8305) 대응
+
+소켓 라이브러리는 이제 [happy eyeballs version 2 (rfc 8305)](https://datatracker.ietf.org/doc/html/rfc8305)를 지원합니다. 이는 `TCPSocket.new` (`TCPSocket.open`)와 `Socket.tcp`에서 네트워크 연결을 효율적이고 안정적으로 제공하기 위해 많은 언어에서 널리 사용되는 방법의 최신 표준화 버전입니다.
+이 개선 사항은 Ruby가 현대적인 인터넷 환경에 적응된 효율적이고 신뢰할 수 있는 네트워크 연결을 제공할 수 있도록 합니다.
+
+Ruby 3.3까지 이러한 메서드는 이름 해결과 연결 시도를 직렬로 수행했습니다. 해당 알고리즘 도입으로 이제 다음과 같이 작동합니다.
+
+1. IPv6와 IPv4 이름 해결을 동시에 수행
+2. IPv6를 우선하며 병렬로 연결을 시도하되 250ms 간격으로 순차적으로 시도
```
[the remark this answers is not in the document]
Comment (riseshia) [reply ko/news/_posts/2024-12-25-ruby-3-4-0-released.md]: 👍

File: ko/news/_posts/2024-12-25-ruby-3-4-0-released.md [hunk 8]
```diff
@@ -0,0 +1,344 @@
...
+
+parse.y에서 Prism으로 기본 파서를 변경했습니다. [[Feature #20564]]
+
+이는 내부적인 개선이며 사용자에게는 거의 변화가 없어야 합니다. 호환성 문제가 발생하면 보고해 주세요.
+
+기존 파서를 사용하려면 `--parser=parse.y` 커맨드 라인 인수를 사용하세요.
+
+## 소켓 라이브러리가 Happy Eyeballs Version 2 (RFC 8305) 대응
+
+소켓 라이브러리는 이제 [happy eyeballs version 2 (rfc 8305)](https://datatracker.ietf.org/doc/html/rfc8305)를 지원합니다. 이는 `TCPSocket.new` (`TCPSocket.open`)와 `Socket.tcp`에서 네트워크 연결을 효율적이고 안정적으로 제공하기 위해 많은 언어에서 널리 사용되는 방법의 최신 표준화 버전입니다.
```
[the remark this answers is not in the document]
Comment (riseshia) [reply ko/news/_posts/2024-12-25-ruby-3-4-0-released.md]: +1 to no space, 2 is the part of RFC title in this context & i cant find the reason we add the space here. Is there some grammer for this?

**Comment by yous** [comment]
#3449

File: ko/news/_posts/2024-12-25-ruby-3-4-0-released.md [hunk 9]
```diff
@@ -0,0 +1,344 @@
...
+* 상세 모드(`-w`)에서 메서드에 넘긴 블록이 사용되지 않았을 때
+  경고를 출력합니다.
+  [[Feature #15554]]
+
+* `String.freeze`나 `Integer#+`처럼 인터프리터와 JIT이 특별히 최적화하는
+  몇몇 코어 메서드를 재정의하면 성능 클래스
+  경고(`-W:performance`나 `Warning[:performance] = true`)를 출력합니다.
+  [[Feature #20429]]
+
+더 자세한 내용은 [NEWS](https://github.com/ruby/ruby/blob/{{ release.tag }}/NEWS.md)나
```
Comment (riseshia) [inline ko/news/_posts/2024-12-25-ruby-3-4-0-released.md:275]:
```suggestion
더 자세한 내용은 [NEWS](https://docs.ruby-lang.org/en/3.4/NEWS_md.html)나
```

https://github.com/ruby/www.ruby-lang.org/pull/3449

**Comment by riseshia** [comment]
Added 460ca9a

**Review by yous — APPROVED** [review]
Thanks!

**Comment by riseshia** [comment]
Thanks for review!!

**Pull request merged**